src/ACF/Block_Config.php

```diff
@@ -97,44 +83,18 @@ public function get_field_group() {
 			'block' => [ static::NAME ],
 		] );
 
-		$group->add_field( $this->get_tab(
-			self::CONTENT_TAB,
-			__( 'Content', 'tribe' )
-		) );
-
-		foreach ( $this->fields as $field ) {
-			$group->add_field( $field );
-		}
-
-		if ( ! $this->settings->count() ) {
-			return $group;
-		}
-
-		$group->add_field( $this->get_tab(
-			self::SETTING_TAB,
-			__( 'Settings', 'tribe' )
-		) );
-
-		foreach ( $this->settings as $setting ) {
-			$group->add_field( $setting );
+		foreach ( $this->items as $block_item ) {
+			if ( $block_item instanceof Field_Section ) {
```

Comment:
`instanceof` strikes me as a code smell indicating that we're not using polymorphism as we should. I understand why you're using it here; ultimately the section is just a field that sits before its "children". By the time we get to `\Tribe\Libs\ACF\Group::get_attributes()`, we would like it to be a field just like all the others.

I think, though, that we might be able to change the behavior of `\Tribe\Libs\ACF\ACF_Configuration::get_attributes()` to something like:

```
public function get_attributes() {
	return [ $this->attributes ];
}
```

Notice the array wrapper around the return value. We would do the same for all the classes currently overriding it (`Repeater`, etc.). But for `Field_Section`, we return an array with the sections attributes and all its children's attributes.

Then in `\Tribe\Libs\ACF\Group::get_attributes()`, we would merge all those arrays:

```
$attributes['fields'] = $this->fields ? array_merge( ... array_map( function( Field $f ) {
	return $f->get_attributes();
}, $this->fields ) ) : [];
```

Now the logic for assembling the aggregate field (the section) is delegated to that field type, just as we do for the other aggregate types.

Reply:
@jbrinley one thing that complicates this approach, is that I was trying to stay flexible with using sections. If someone registers fields outside of a section then the `fields` attributes gets a list of field arrays wrapped in an extra `[]`

Think I'll take the weekend to think about a better approach but welcome suggestions.

Reply:
That's what the `array_merge` accomplishes. It can take an array with a single field in it (what you would get when registering fields outside of sections) or an array with multiple fields (as in a section).

Reply:
I think I needed to take the parent::get_attributes() calls and just do $this->attributes to not have the extra `[]`